src/api/app.py

```diff
@@ -185,19 +186,18 @@ async def on_accept(self, socket: WebSocket, api_key: str, session_id: str) -> N
         self.api_key = api_key
         self.session_id = session_id
         self.path = _make_demo_path(self.session_id)
-        self.handle = open(self.path, "wb")
+
+        demo_path_exists = os.path.exists(self.path)
+        if demo_path_exists:
+            mode = "ab"
+        else:
+            mode = "wb"
+        self.handle = open(self.path, mode)
```

Comment:
if the demo file exists and we connect to the socket, this implies that we never explicitly closed the session, which the client handles in all cases except a ctrl c